modules/convert/convert.go

```diff
@@ -414,3 +415,25 @@ func ToLFSLock(l *git_model.LFSLock) *api.LFSLock {
 		},
 	}
 }
+
+// ToChangedFile convert a gitdiff.DiffFile to api.ChangedFile
+func ToChangedFile(f *gitdiff.DiffFile, repo *repo_model.Repository, commit string) *api.ChangedFile {
+	status := "changed"
+	if f.IsDeleted {
+		status = "deleted"
+	}
+	file := &api.ChangedFile{
+		NameHash:  f.NameHash,
+		Filename:  f.Name,
+		Status:    status,
+		Additions: f.Addition,
+		Deletions: f.Deletion,
+	}
+
+	if f.Name != "" {
+		file.HTMLURL = fmt.Sprint(repo.HTMLURL(), "/src/commit/", commit, "/", f.GetDiffFileName())
+		file.ContentsURL = fmt.Sprint(repo.HTMLURL(), "/raw/commit/", commit, "/", f.GetDiffFileName())
+	}
```

Comment:
2 problems here:

1. The diff file name needs to be encoded, otherwise there will be invalid URLs.
2. After reading the logic of `GetDiffFileName`, I think there the `f.Name` should be used directly instead of the `GetDiffFileName`